product/internal/proto_files/domain/product.proto

```diff
@@ -7,15 +7,25 @@ package product.internal.proto_files.domain;
 import "google/protobuf/timestamp.proto";
 
 /* set where the generate Go code will be placed */
-option go_package = "internal/grpc/domain_service";
+option go_package = "internal/grpc/domain";
 
 message Product {
-  string id = 1;
+  // @inject_tag: gorm:"primary_key"
+  uint64 id = 1;
```

Comment:
This is OK at small scale and when using a database that requires ints as keys, but when scaling a database horizontally it's difficult to ensure uniqueness of primary keys. Often a UUID is used.

Keep in mind that I'm absolutely not an expert in databases!

Reply:
I will work on changing the keys to UUID, it definitely has more advantages than using regular IDs.